mls-rs/src/group/mod.rs

```diff
@@ -1734,10 +1734,10 @@ where
         self.state_repo.insert(past_epoch).await?;
 
         self.epoch_secrets = key_schedule_result.epoch_secrets;
-        self.state.context = provisional_state.group_context;
+        self.state.context = provisional_state.group_context.clone();
         self.state.interim_transcript_hash = interim_transcript_hash;
         self.key_schedule = key_schedule_result.key_schedule;
-        self.state.public_tree = provisional_state.public_tree;
+        self.state.public_tree = provisional_state.public_tree.clone();
```

Comment:
AFAIR these clones were a bit expensive for large trees (messing up benchmark results). Would it be worth considering to move `public_tree` out of `provisional_state`, placing an empty tree in there -- the state update can use the tree from `self.state` to compute stuff if needed?